File: cpp/src/mono/Result.cpp
#include "Result.hpp"

#include <iostream>
#include <algorithm>

namespace mono {
    Result::Result(vector<double> solution) {
        setSolution(solution);
        setCost(0);
    }
    
    Result::Result(vector<double> solution, double cost) {
        setSolution(solution);
        setCost(cost);
    }

    Result::Result(vector<double> solution, double cost, vector<double> history) {
        setSolution(solution);
        setCost(cost);
        setHistory(history);
    }

    void Result::displaySolution() const {
        std::cout << "Cost     : " << cost << std::endl << "Solution : ";
        std::for_each(solution.begin(), solution.end(), [] (double value) {
            std::cout << value << "  ";
        });
        std::cout << std::endl;
    }

};

File: java/joptool/src/main/java/com/gritti/joptool/mono/Result.java
package com.gritti.joptool.mono;

public class Result {
	private final double[] solution;
	private final double[] history;

	public Result(double[] solution, double[] history) {
		this.solution = solution.clone();
		this.history = history.clone();
	}

	public double[] getSolution() {
		return solution.clone();
	}

	public double[] getHistory() {
		return history.clone();
	}

	public double getCost() {
		return history[history.length - 1];
	}
}
